src/index.tsx

```diff
@@ -42,108 +29,12 @@ export type UseModal = (
   isOpen: boolean
 ];
 
-const wrapperStyle: React.CSSProperties = {
-  position: 'fixed',
-  top: 0,
-  left: 0,
-  bottom: 0,
-  right: 0,
-  display: 'flex',
-  justifyContent: 'center',
-  alignItems: 'center',
-  zIndex: 1000,
-};
-
-const overlayStyle: React.CSSProperties = {
-  position: 'fixed',
-  top: 0,
-  left: 0,
-  bottom: 0,
-  right: 0,
-  backgroundColor: 'rgba(0, 0, 0, 0.5)',
-  zIndex: 100000,
-};
-
-const containerStyle: React.CSSProperties = {
-  position: 'relative',
-  zIndex: 100001,
-};
-
-const closeButtonStyle: React.CSSProperties = {
-  // reset
-  backgroundColor: 'transparent',
-  border: 'none',
-  cursor: 'pointer',
-  padding: 0,
-  appearance: 'none',
-
-  position: 'absolute',
-  right: 0,
-  top: 0,
-  width: '40px',
-  height: '40px',
-  display: 'flex',
-  alignItems: 'center',
-  justifyContent: 'center',
-  fontSize: '28px',
-  color: '#fff',
-  transform: 'translateX(100%)',
-};
-
-const DefaultCloseButton: React.FC<DefaultCloseButtonProps> = ({ onClose }) => {
-  return (
-    <button
-      type="button"
-      style={closeButtonStyle}
-      onClick={onClose}
-      aria-label="close"
-    >
-      ×
-    </button>
-  );
-};
-
-const Modal: React.FC<ModalProps> = ({
-  children,
-  isOpen,
-  close,
-  elementId = 'root',
-  preventScroll,
-  focusTrapOptions,
-  closeButton,
-}) => {
-  const dialogRef = useRef<HTMLDivElement>(null);
-  useFocusTrap(dialogRef, isOpen, {
-    onDeactivate: close,
-    clickOutsideDeactivates: true,
-    ...focusTrapOptions,
-  });
-  useBodyScrollLock(dialogRef, isOpen, preventScroll);
-
-  if (isOpen === false) {
-    return null;
-  }
-
-  return createPortal(
-    <div style={wrapperStyle}>
-      <div style={overlayStyle} />
-      <div
-        ref={dialogRef}
-        role="dialog"
-        aria-modal="true"
-        style={containerStyle}
-        tabIndex={-1}
-      >
-        {children}
-        {closeButton}
-      </div>
-    </div>,
-    document.getElementById(elementId) as HTMLElement
-  );
-};
-
 export const useModal: UseModal = (elementId = 'root', options = {}) => {
-  const { preventScroll = false, focusTrapOptions = {}, ...rest } = options;
+  const {
+    preventScroll = false,
+    focusTrapOptions = {},
+    ...rest
+  } = deepmerge<ModalOptions>(useModalConfig(), options);
```

Comment:
Because `focusTrapOptions` are objects, each must be overridden with respect to its value.